backend/MASZ/Controllers/api/v1/VerifiedEvidenceController.cs

```diff
@@ -0,0 +1,95 @@
+﻿using MASZ.Enums;
+using MASZ.Repositories;
+using MASZ.Models.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using MASZ.Models.Views;
+using MASZ.Dtos.VerifiedEvidence;
+
+namespace MASZ.Controllers.api.v1
+{
+    [ApiController]
+    [Route("api/v1/guilds/{guildId}/evidence")]
+    [Authorize]
+    public class VerifiedEvidenceController : SimpleController
+    {
+        public VerifiedEvidenceController(IServiceProvider serviceProvider) : base(serviceProvider)
+        {
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllEvidence([FromRoute] ulong guildId, [FromQuery] int startPage = 0)
+        {
+            await RequirePermission(guildId, DiscordPermission.Moderator);
+            List<VerifiedEvidenceView> evidence = (await VerifiedEvidenceRepository.CreateDefault(_serviceProvider, await GetIdentity()).GetEvidencePagination(guildId, startPage)).Select(x => new VerifiedEvidenceView(x)).ToList();
+            if(evidence.Count == 0)
+            {
+                return NotFound();
+            }
+            return Ok(evidence);
+        }
+
+        [HttpGet("{evidenceId}")]
+        public async Task<IActionResult> GetEvidence([FromRoute] ulong guildId, [FromRoute] int evidenceId)
+        {
+            await RequirePermission(guildId, DiscordPermission.Moderator);
+            VerifiedEvidenceView evidence = new(await VerifiedEvidenceRepository.CreateDefault(_serviceProvider, await GetIdentity()).GetEvidence(guildId, evidenceId));
+            if (evidence == default)
```

Comment:
the method above should throw `ResourceNotFoundException`, that way the 404 case is already handled.

Reply:
Yeah that makes more sense. Will do